File: 剑指offer/14- I. 剪绳子.py
class Solution:
    def cuttingRope(self, n: int) -> int:
        if n <= 3: return n - 1
        a, b = n // 3, n % 3
        if b == 0: return int(math.pow(3, a))
        if b == 1: return int(math.pow(3, a - 1) * 4) # 这里写错过，改成了elif就报错
        return int(math.pow(3, a) * 2)
    
#时间复杂度：O(1)
#空间复杂度：O(1)

解法：
https://leetcode-cn.com/problems/jian-sheng-zi-lcof/solution/mian-shi-ti-14-i-jian-sheng-zi-tan-xin-si-xiang-by/
证明：
https://leetcode-cn.com/problems/integer-break/solution/zheng-shu-chai-fen-shu-xue-fang-fa-han-wan-zheng-t/

File: 剑指offer/C++_Version/14- I. 剪绳子.cpp
//方法1

class Solution {
public:
    int cuttingRope(int n) {
        if(n<=3) return n-1;
        int a = n / 3;
        int b = n % 3;
        if(b==0) 
            return pow(3, a);
        if(b==1) return 4*pow(3, a-1);
        else return 2*pow(3, a);
    }
};
